src/oemof/tabular/postprocessing/core.py

```diff
@@ -0,0 +1,238 @@
+    calculation: Type["Calculation"]
+    parameters: Optional[dict] = None
+
+
+def get_dependency_name(
+    calculation: Union[
+        "Calculation", Type["Calculation"], ParametrizedCalculation
+    ]
+):
+    if isinstance(calculation, Calculation):
+        # Get name from instance
+        signiture = inspect.signature(calculation.__init__)
+        return "_".join(
+            [calculation.name]
+            + [
+                f"{parameter}={getattr(calculation, parameter)}"
+                for parameter in signiture.parameters
+                if parameter not in ("self", "calculator")
+            ]
+        )
+    # Get name from class and default parameters in class
+    if isinstance(calculation, ParametrizedCalculation):
+        calc = calculation.calculation
+        parameters = calculation.parameters or {}
+    else:
+        calc = calculation
+        parameters = {}
+    signiture = inspect.signature(calc.__init__)
+    names = [calc.name]
+    for name, sig_parameter in signiture.parameters.items():
+        if name not in ("self", "calculator"):
+            value = parameters.get(name)
+            if value:
+                names.append(f"{name}={value}")
+                continue
+            if sig_parameter.default is inspect.Parameter.empty:
+                raise CalculationError(
+                    f"Parameter '{name}' in calculation '{calc.name}' not set."
+                )
+            names.append(f"{name}={sig_parameter.default}")
+    return "_".join(names)
+
+
+class Calculator:
+    """Entity to gather calculations and their results"""
+
+    def __init__(self, input_parameters, output_parameters):
+        self.calculations = {}
+        self.scalar_params = self.__init_df(input_parameters, "scalars")
+        self.scalars = self.__init_df(output_parameters, "scalars")
+        self.sequences_params = self.__init_df(input_parameters, "sequences")
+        self.sequences = self.__init_df(output_parameters, "sequences")
+        self.busses = self.__filter_type("bus")
+        self.links = self.__filter_type("link")
+        logging.info("Successfully set up calculator")
+
+    @staticmethod
+    def __init_df(oemof_data, data_type="scalars"):
+        r"""
+        Converts scalars/sequences dictionary to a multi-indexed
+        DataFrame.
+        """
+        data = {
+            tuple(str(k) if k is not None else None for k in key): (
+                value[data_type]
+                if isinstance(
+                    value[data_type],
+                    pd.Series if data_type == "scalars" else pd.DataFrame,
+                )
+                else (
+                    pd.Series(value[data_type], dtype="object")
+                    if data_type == "scalars"
+                    else pd.DataFrame.from_dict(
+                        value[data_type], dtype="object"
+                    )
+                )
+            )
+            for key, value in oemof_data.items()
+        }
```

Comment:
Started to fix multi-period results in postprocessing.
Most functions can stay the same - only result scalars have to be changed, as they are delivered as DataFrames (instead of Series) with periods as columns.
But when trying to test multi-period investment, I saw that datapackage results of `investment_multi_period` are broken (only three result rows are returned, but we have multiple components in place?!) could you check, @nailend ?
Once investment datapackage gives correct results - I can test/fix my postprocessing module.